tests/anderson_test.py

```diff
@@ -114,8 +125,7 @@ def f(x, r, y):
     sol, state = aa.init(x0, r, y0)
     sols = []
     for i in range(10):
-      _, y = state.value
-      sol, state = aa.update(sol, state, r, y)
+      sol, state = aa.update(sol, state, r, state.aux)
```

Comment:
I noticed a similar test for `FixedPointIteration` and was a bit confused by the fact that `state.aux` is used to feed extra arguments to the function. This means that it wouldn't be possible to do the equivalent with `run`. Normally, the extra arguments are supposed to be constant. And the argument we iterate over is supposed to be the first one.

Reply:
I understand. The example has been modified.